qiskit/transpiler/passes/optimization/optimize_1q_decomposition.py

```diff
@@ -147,62 +201,42 @@ def run(self, dag):
                 basis = self._target.operation_names_for_qargs((qubit,))
 
             if new_dag is not None and self._substitution_checks(dag, run, new_dag, basis, qubit):
+                new_dag = self._gate_sequence_to_dag(new_dag)
```

Comment:
Possibly let's rename the outer `new_dag` if it's not actually a DAG any more.

Reply:
Done in: https://github.com/Qiskit/qiskit-terra/pull/9578/commits/89db1da2d797475094449d93a2e6eb043db58fae